SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialRPCService.h

```diff
@@ -20,6 +20,9 @@ DECLARE_DELEGATE_RetVal_ThreeParams(bool, ExtractRPCDelegate, Worker_EntityId, E
 
 namespace SpatialGDK
 {
+// The RPC must have either succeeded with no subsequent action to take, or failed with an indicator of the action to take
+#define VALIDATE_RPC_PUSH_RESULT(Result) check((PushRPCResultUtils::GetOutcome(Result) == EPushRPCResult::Success && !PushRPCResultUtils::IsAnyFlagSet(Result, EPushRPCResult::AllFailureActions)) || \
```

Comment:
Does this have to be a macro? I'd prefer it as an inline function.

Reply:
+1

Reply:
Yep I can make that change